### PR TITLE
Has many metadata

### PR DESCRIPTION
**Pull request by wecc**

### CODE REVIEW

**Comment by wecc** [comment]
This should be good to merge.

File: packages/ember-data/lib/system/clone-null.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function cloneNull(source) {
```
Comment (stefanpenner) [inline packages/ember-data/lib/system/clone-null.js]: export default function cloneNull

Comment (wecc) [inline packages/ember-data/lib/system/clone-null.js]: Corrected, thanks!

File: packages/ember-data/lib/system/relationships/state/has-many.js [hunk 2]
```diff
@@ -144,6 +145,9 @@ ManyRelationship.prototype.computeChanges = function(records) {
 ManyRelationship.prototype.fetchLink = function() {
   var self = this;
   return this.store.findHasMany(this.record, this.link, this.relationshipMeta).then(function(records) {
+    var meta = self.store.metadataFor(self.relationshipMeta.type);
+    self.manyArray.set('meta', cloneNull(meta));
```
Comment (igorT) [inline packages/ember-data/lib/system/relationships/state/has-many.js]: We have a potential problem here, if you make a request for a record, set some metadata, and then fetch the link, *do not set the metadata*, this array will end up with wrong metadata. Not sure how to fix this right now

**Comment by wecc** [comment]
@igorT Rebased and merge conflicts fixed, should be good to go. Moved failing test to #2996.

**Comment by igorT** [comment]
image
such lies

**Comment by igorT** [comment]
Superseded by #3221

**Pull request closed without merging**